// eslint-plugin-lit/lib/rules/no-useless-template-literals.js
"use strict";
/**
 * @fileoverview Disallows redundant literal values in templates
 * @author James Garbutt <https://github.com/43081j>
 */
//------------------------------------------------------------------------------
// Rule Definition
//------------------------------------------------------------------------------
const rule = {
    meta: {
        docs: {
            description: 'Disallows redundant literal values in templates',
            category: 'Best Practices',
            recommended: true,
            url: 'https://github.com/43081j/eslint-plugin-lit/blob/master/docs/rules/no-useless-template-literals.md'
        }
    },
    create(context) {
        // variables should be defined here
        //----------------------------------------------------------------------
        // Helpers
        //----------------------------------------------------------------------
        //----------------------------------------------------------------------
        // Public
        //----------------------------------------------------------------------
        return {
            TaggedTemplateExpression: (node) => {
                if (node.type === 'TaggedTemplateExpression') {
                    for (const expr of node.quasi.expressions) {
                        if (expr.type === 'Literal') {
                            context.report({
                                node: expr,
                                message: 'Literals must not be substituted into templates'
                            });
                        }
                    }
                }
            }
        };
    }
};
module.exports = rule;
